### ISSUE TITLE
Support Node v10.x LTS

### ISSUE THREAD

**Issue by jwflory** [RITlug/teleirc#99]
# Summary

Officially support Node v10.x LTS and integrate support into CI tests

# Background

NodeJS upstream released Node v10.0.0 in [April 2018](https://nodejs.org/en/download/releases/). It's had a few months in the wild and will receive long-term support for a few extra years. It's a good idea to keep up with upstream LTS so we don't fall behind. Only supporting LTS releases is also easier for the personal bandwidth of everyone involved with development.

# Details

Updating Travis CI tests to also run with Node v10.x will close this ticket.

Currently, the bot is running without issues in a Node v10.14.1 environment. We need to make sure our unit tests still work tested up to this version.

# Outcome

Better support for newer versions of Node, more cookie points for users / developers who want to use newer versions of Node

**Comment by Tjzabel** [comment]
I agree. It's also worth noting that NodeJS ~8 is reaching the end of its "active" LTS cycle January 1st, 2019, and moving into a "maintenance" cycle. 

It would definitely be a good idea to stay up to date with the current, active LTS release.

**Comment by jwflory** [comment]
@Tjzabel Nice catch. Given the size of our development community, I wonder if it is a good idea to make a hard switch to Node v10.x LTS and for us to "drop" support for v8.x. Or should we try to offer official support for both v8 and v10 LTS?

Given my general inexperience with Node, I'm not sure what the best call is here.

**Comment by Tjzabel** [comment]
At the very moment, we could easily support both. 
Unless we start implementing features only found in Node v10, it would be trivial to support both versions. 

It might be a good idea to see what version of Node comes standard with package managers. Even if a hard switch to v10 LTS is fine, I wouldn't want to only support a release that is too recent for apt/yum/dnf to have in their repositories, as that would make things a little more difficult for those who want to run teleirc.

**Comment by jwflory** [comment]
@Tjzabel Fair. I don't anticipate us using any new features right away. For Fedora, they ship v8 but this could change in Fedora 30 or 31. Not sure about other distributions. We can arbitrarily support this via CI.

To close this issue, I suggest two actions:

1. Update Travis tests to run against v8 + v10
2. Update documentation for officially supported versions
